sdk/python/kfp/_client.py

```diff
@@ -1179,21 +1211,52 @@ def list_recurring_runs(self,
                 filter=filter)
         return response
 
-    def get_recurring_run(self, job_id):
+    def get_recurring_run(self,
+                          job_id: str = None,
+                          job_name: str = None,
+                          namespace: str = None) -> kfp_server_api.ApiJob:
         """Get recurring_run details.
 
         Args:
           job_id: id of the recurring_run.
+          job_name: name of the recurring_run.
 
         Returns:
           A response object including details of a recurring_run.
 
-        Throws:
-          Exception if recurring_run is not found.
+        Raises:
+          ValueError: If required fields are not supplied.
+          kfp_server_api.ApiException: If recurring_run is not found.
         """
-        return self._job_api.get_job(id=job_id)
+        namespace = namespace or self.get_user_namespace()
+        if job_id is None and job_name is None:
+            raise ValueError('Either job_id or job_name is required')
+        if job_id is not None:
+            return self._job_api.get_job(id=job_id)
+
+        recurring_run_filter = json.dumps({
```

Comment:
Again, I would suggest keep the get method simple and straightforward--just retrieve by id not filtering by name.

Reply:
Yeah in general I'd agree it's not the nicest. To take a step back, what I was trying to achieve with this was to make a way for the CLI to query for recurring runs, pipelines, etc. by name conveniently, do you have a recommended way forward for achieving that? 

I know technically I could use the CLI client to `list` all the results then search through that list by name, but then we'd need to deal with pagination etc. so I thought instead making the `client` methods accept either an ID or a name, rework the CLI interfaces to conform that optionality would be a slightly nicer user experience for CLI users but also by pushing it down the `client` it'd also make it nicer for users of the SDK itself (assuming we can make the changes backwards compatible).

---

Some more background on why getting by name is important. We are using the `kfp` CLI during CICD (whether we should is a great topic for discussion too but we are for now at least!), so we construct the names of resources (ie. pipelines, recurring runs, etc.) dynamically based on the project that we're deploying. For resources that live beyond individual CICD pipeline runs (ie. across commits) this means we can mutate existing resources that live between deploys.

Reply:
There's already a `list_recurring_runs` method in the client.
https://github.com/kubeflow/pipelines/blob/da77ef554e5ed0d8a6148fdc1a3886d7ae1125c2/sdk/python/kfp/_client.py#L1141-L1146
It feels to me that getting recurring run by name can be a simple call of that method. 

> I know technically I could use the CLI client to list all the results then search through that list by name, but then we'd need to deal with pagination etc.

The CLI doesn't expose the filter option? If not, sounds like we should add that to the CLI interface. WDYT?

Reply:
Ah yeah that makes way more sense, thanks for the suggestion and putting me on the right track. Let me rework the PR to adopt that convention across the touch points I've changed and I'll re-request your review when I've got things aligned in that direction 😄